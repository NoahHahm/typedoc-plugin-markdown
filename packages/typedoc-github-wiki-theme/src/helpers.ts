export function parseUrl(url: string) {
  return encodeURI('../wiki/' + url.replace('.md', ''));
}
