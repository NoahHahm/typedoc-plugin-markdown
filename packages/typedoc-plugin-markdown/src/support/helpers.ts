import {
  ContainerReflection,
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
  SignatureReflection,
} from 'typedoc';
import { backTicks } from './els';

/**
 * Determines if current signature is a constructor
 */
export const isConstructor = (
  signature: DeclarationReflection | SignatureReflection,
) => {
  return signature.parent?.kindOf(ReflectionKind.Constructor);
};

export function getIndexHeadingLevel(
  reflection: DeclarationReflection | ProjectReflection,
) {
  if (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) ||
    reflection.hasOwnDocument
  ) {
    return 2;
  }
  return 4;
}

export function getGroupHeadingLevel(container: ContainerReflection) {
  return container.hasOwnDocument ? 2 : 4;
}

export function getReflectionHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  if (reflection.hasOwnDocument) {
    return 1;
  }
  if (reflection.kindOf(ReflectionKind.Constructor)) {
    return reflection.parent?.hasOwnDocument ? 2 : 4;
  }
  return reflection.parent?.hasOwnDocument ? 3 : 5;
}

export function getMemberHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  return getReflectionHeadingLevel(reflection) + 1;
}

export function getReflectionTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  md.push(reflection.name);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    md.push(`\\<${typeParameters}\\>`);
  }

  return md.join('');
}

export function getFlags(reflection: DeclarationReflection) {
  if (reflection.flags?.length > 0 && !reflection.flags.isRest) {
    return reflection.flags.map((flag) => backTicks(flag)).join(' ');
  }
  return null;
}
