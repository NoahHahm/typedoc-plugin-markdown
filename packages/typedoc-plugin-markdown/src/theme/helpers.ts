/**
 *
 * A set of helpers that parses TypeDoc models to be consumed by theme resources.
 *
 * @module
 */

import {
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
} from 'typedoc';
import { backTicks } from '../support/elements';
import { escapeChars } from '../support/utils';

export function getDeclarationType(declaration: DeclarationReflection) {
  if (declaration.getSignature) {
    return declaration.getSignature.type;
  }
  if (declaration.setSignature) {
    return declaration.setSignature.type;
  }
  return declaration.type;
}

export function getProjectDisplayName(
  project: ProjectReflection,
  includeVersion: boolean,
): string {
  const version =
    includeVersion && project.packageVersion
      ? ` - v${project.packageVersion}`
      : '';
  return `${project.name}${version ? version : ''}`;
}

export function hasIndex(
  reflection: DeclarationReflection | ProjectReflection,
) {
  return reflection.groups?.some((group) => group.allChildrenHaveOwnDocument());
}

export function hasTOC(reflection: DeclarationReflection | ProjectReflection) {
  return (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) &&
    reflection.groups?.some((group) => !group.allChildrenHaveOwnDocument())
  );
}

export function isGroupKind(reflection: DeclarationReflection) {
  return reflection.kindOf([
    ReflectionKind.Class,
    ReflectionKind.Interface,
    ReflectionKind.Enum,
    ReflectionKind.Function,
    ReflectionKind.Variable,
    ReflectionKind.TypeAlias,
  ]);
}

export function getModifier(reflection: DeclarationReflection) {
  if (reflection.flags.isAbstract) {
    return 'abstract';
  }
  if (reflection.flags.isPrivate) {
    return 'private';
  }
  if (reflection.flags.isReadonly) {
    return 'readonly';
  }
  if (reflection.flags.isStatic) {
    return 'static';
  }
  if (reflection.getSignature) {
    return 'get';
  }
  if (reflection.setSignature) {
    return 'set';
  }
  return null;
}

export const KEYWORD_MAP = {
  [ReflectionKind.Class]: 'class',
  [ReflectionKind.Interface]: 'interface',
  [ReflectionKind.Enum]: 'enum',
  [ReflectionKind.TypeAlias]: 'type',
};

export function getMemberTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  const name: string[] = [];

  const flagsToInclude = ['Abstract'];

  const flags = reflection.flags.filter((flag) =>
    flagsToInclude.includes(flag),
  );

  if (flags.length) {
    name.push(flags.map((flag) => backTicks(flag.toLowerCase())).join(' '));
    name.push(' ');
  }

  name.push(`${escapeChars(reflection.name)}`);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    name.push(`${backTicks(`<${typeParameters}>`)}`);
  }

  md.push(name.join(''));

  return md.join(': ');
}
