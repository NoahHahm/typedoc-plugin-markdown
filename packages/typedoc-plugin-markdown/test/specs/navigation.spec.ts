import {
  FixtureOutputDir,
  FixtureOutputFileStrategy,
} from '../__utils__/config';
import { expectFileToEqual } from '../__utils__/helpers';

describe(`Navigation`, () => {
  test(`should gets Navigation Json for single entry point`, () => {
    expectFileToEqual(
      FixtureOutputDir.Reflections,
      [FixtureOutputFileStrategy.Members, FixtureOutputFileStrategy.Modules],
      'sidebar.json',
      1,
    );
  });

  test(`should gets Navigation Json for multiple entry points`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      [FixtureOutputFileStrategy.Members, FixtureOutputFileStrategy.Modules],
      'sidebar.json',
    );
  });

  test(`should get Navigation Json for packages`, () => {
    expectFileToEqual(
      FixtureOutputDir.Packages,
      [FixtureOutputFileStrategy.Members, FixtureOutputFileStrategy.Modules],
      'sidebar.json',
      1,
    );
  });
});
