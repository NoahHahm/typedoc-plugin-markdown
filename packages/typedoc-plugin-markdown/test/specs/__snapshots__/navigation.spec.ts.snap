// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should get Navigation Json for packages: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "modules.md"
  },
  {
    "title": "package-1",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-1/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-1/modules.md"
  },
  {
    "title": "package-2",
    "children": [
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "PackageInterface",
            "url": "package-2/interfaces/PackageInterface.md"
          }
        ]
      }
    ],
    "url": "package-2/modules.md"
  }
]"
`;

exports[`Navigation should get Navigation Json for packages: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "modules.md"
  },
  {
    "title": "package-1",
    "children": [],
    "url": "package-1/modules.md"
  },
  {
    "title": "package-2",
    "children": [],
    "url": "package-2/modules.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "modules.md"
  },
  {
    "title": "basic",
    "url": "basic/README.md",
    "children": [
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "EnumA",
            "url": "basic/enumerations/EnumA.md"
          },
          {
            "title": "EnumB",
            "url": "basic/enumerations/EnumB.md"
          }
        ]
      },
      {
        "title": "Classes",
        "children": [
          {
            "title": "ClassA",
            "url": "basic/classes/ClassA.md"
          },
          {
            "title": "ClassB",
            "url": "basic/classes/ClassB.md"
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "IterfaceA",
            "url": "basic/interfaces/IterfaceA.md"
          },
          {
            "title": "IterfaceB",
            "url": "basic/interfaces/IterfaceB.md"
          }
        ]
      },
      {
        "title": "Type Aliases",
        "children": [
          {
            "title": "TypeA",
            "url": "basic/type-aliases/TypeA.md"
          },
          {
            "title": "TypeB",
            "url": "basic/type-aliases/TypeB.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-categories",
    "url": "has-categories/README.md",
    "children": [
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "Category A",
            "children": [
              {
                "title": "CategoryAEnum1",
                "url": "has-categories/enumerations/CategoryAEnum1.md"
              },
              {
                "title": "CategoryAEnum2",
                "url": "has-categories/enumerations/CategoryAEnum2.md"
              }
            ]
          },
          {
            "title": "Category B",
            "children": [
              {
                "title": "CategoryBEnum1",
                "url": "has-categories/enumerations/CategoryBEnum1.md"
              },
              {
                "title": "CategoryBEnum2",
                "url": "has-categories/enumerations/CategoryBEnum2.md"
              }
            ]
          },
          {
            "title": "Other",
            "children": [
              {
                "title": "UnCategorizedEnum",
                "url": "has-categories/enumerations/UnCategorizedEnum.md"
              }
            ]
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "Category A",
            "children": [
              {
                "title": "CategoryAInterface1",
                "url": "has-categories/interfaces/CategoryAInterface1.md"
              },
              {
                "title": "CategoryAInterface2",
                "url": "has-categories/interfaces/CategoryAInterface2.md"
              }
            ]
          },
          {
            "title": "Category B",
            "children": [
              {
                "title": "CategoryBInterface1",
                "url": "has-categories/interfaces/CategoryBInterface1.md"
              },
              {
                "title": "CategoryBInterface2",
                "url": "has-categories/interfaces/CategoryBInterface2.md"
              }
            ]
          },
          {
            "title": "Other",
            "children": [
              {
                "title": "UnCategorizedInterace",
                "url": "has-categories/interfaces/UnCategorizedInterace.md"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/README.md",
    "children": [
      {
        "title": "Namespaces",
        "children": [
          {
            "title": "NamespaceB",
            "url": "has-namespaces/namespaces/NamespaceB/README.md"
          },
          {
            "title": "_Namespace_A_",
            "url": "has-namespaces/namespaces/Namespace_A_/README.md",
            "children": [
              {
                "title": "Namespaces",
                "children": [
                  {
                    "title": "NestedNamespace",
                    "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/README.md",
                    "children": [
                      {
                        "title": "Interfaces",
                        "children": [
                          {
                            "title": "NamespaceInterface",
                            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md"
                          }
                        ]
                      }
                    ]
                  }
                ]
              },
              {
                "title": "Interfaces",
                "children": [
                  {
                    "title": "NamespaceInterface",
                    "url": "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md"
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        "title": "Enumerations",
        "children": [
          {
            "title": "EnumB",
            "url": "has-namespaces/enumerations/EnumB.md"
          },
          {
            "title": "_Enum_A_",
            "url": "has-namespaces/enumerations/Enum_A_.md"
          }
        ]
      },
      {
        "title": "Classes",
        "children": [
          {
            "title": "ClassB",
            "url": "has-namespaces/classes/ClassB.md"
          },
          {
            "title": "__Class_A_",
            "url": "has-namespaces/classes/Class_A_.md"
          }
        ]
      },
      {
        "title": "Interfaces",
        "children": [
          {
            "title": "InterfaceB",
            "url": "has-namespaces/interfaces/InterfaceB.md"
          },
          {
            "title": "_Interface_A_",
            "url": "has-namespaces/interfaces/Interface_A_.md"
          }
        ]
      },
      {
        "title": "Type Aliases",
        "children": [
          {
            "title": "TypeB",
            "url": "has-namespaces/type-aliases/TypeB.md"
          },
          {
            "title": "_Type_A_",
            "url": "has-namespaces/type-aliases/Type_A_.md"
          }
        ]
      },
      {
        "title": "Variables",
        "children": [
          {
            "title": "_variable_A_",
            "url": "has-namespaces/variables/variable_A_.md"
          },
          {
            "title": "variableB",
            "url": "has-namespaces/variables/variableB.md"
          }
        ]
      },
      {
        "title": "Functions",
        "children": [
          {
            "title": "_function_A_",
            "url": "has-namespaces/functions/function_A_.md"
          },
          {
            "title": "functionB",
            "url": "has-namespaces/functions/functionB.md"
          }
        ]
      }
    ]
  },
  {
    "title": "index",
    "url": "index/README.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "members") (Option Group "2") 1`] = `
"[
  {
    "title": "API",
    "url": "index.md"
  },
  {
    "title": "basic",
    "url": "basic/index.md",
    "children": [
      {
        "title": "EnumA",
        "url": "basic/enumerations/EnumA.md"
      },
      {
        "title": "EnumB",
        "url": "basic/enumerations/EnumB.md"
      },
      {
        "title": "ClassA",
        "url": "basic/classes/ClassA.md"
      },
      {
        "title": "ClassB",
        "url": "basic/classes/ClassB.md"
      },
      {
        "title": "IterfaceA",
        "url": "basic/interfaces/IterfaceA.md"
      },
      {
        "title": "IterfaceB",
        "url": "basic/interfaces/IterfaceB.md"
      }
    ]
  },
  {
    "title": "has-categories",
    "url": "has-categories/index.md",
    "children": [
      {
        "title": "Category A",
        "children": [
          {
            "title": "CategoryAEnum1",
            "url": "has-categories/enumerations/CategoryAEnum1.md"
          },
          {
            "title": "CategoryAEnum2",
            "url": "has-categories/enumerations/CategoryAEnum2.md"
          },
          {
            "title": "CategoryAInterface1",
            "url": "has-categories/interfaces/CategoryAInterface1.md"
          },
          {
            "title": "CategoryAInterface2",
            "url": "has-categories/interfaces/CategoryAInterface2.md"
          }
        ]
      },
      {
        "title": "Category B",
        "children": [
          {
            "title": "CategoryBEnum1",
            "url": "has-categories/enumerations/CategoryBEnum1.md"
          },
          {
            "title": "CategoryBEnum2",
            "url": "has-categories/enumerations/CategoryBEnum2.md"
          },
          {
            "title": "CategoryBInterface1",
            "url": "has-categories/interfaces/CategoryBInterface1.md"
          },
          {
            "title": "CategoryBInterface2",
            "url": "has-categories/interfaces/CategoryBInterface2.md"
          }
        ]
      },
      {
        "title": "Other",
        "children": [
          {
            "title": "UnCategorizedEnum",
            "url": "has-categories/enumerations/UnCategorizedEnum.md"
          },
          {
            "title": "UnCategorizedInterace",
            "url": "has-categories/interfaces/UnCategorizedInterace.md"
          }
        ]
      }
    ]
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "NamespaceB",
        "url": "has-namespaces/namespaces/NamespaceB/index.md"
      },
      {
        "title": "_Namespace_A_",
        "url": "has-namespaces/namespaces/Namespace_A_/index.md",
        "children": [
          {
            "title": "NestedNamespace",
            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/index.md",
            "children": [
              {
                "title": "NamespaceInterface",
                "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md"
              }
            ]
          },
          {
            "title": "NamespaceInterface",
            "url": "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md"
          }
        ]
      },
      {
        "title": "EnumB",
        "url": "has-namespaces/enumerations/EnumB.md"
      },
      {
        "title": "_Enum_A_",
        "url": "has-namespaces/enumerations/Enum_A_.md"
      },
      {
        "title": "ClassB",
        "url": "has-namespaces/classes/ClassB.md"
      },
      {
        "title": "__Class_A_",
        "url": "has-namespaces/classes/Class_A_.md"
      },
      {
        "title": "InterfaceB",
        "url": "has-namespaces/interfaces/InterfaceB.md"
      },
      {
        "title": "_Interface_A_",
        "url": "has-namespaces/interfaces/Interface_A_.md"
      }
    ]
  },
  {
    "title": "index",
    "url": "module.index/index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "modules.md"
  },
  {
    "title": "basic",
    "url": "basic.md"
  },
  {
    "title": "has-categories",
    "url": "has-categories.md"
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/README.md",
    "children": [
      {
        "title": "Namespaces",
        "children": [
          {
            "title": "NamespaceB",
            "url": "has-namespaces/namespaces/NamespaceB.md"
          },
          {
            "title": "_Namespace_A_",
            "url": "has-namespaces/namespaces/Namespace_A_/README.md",
            "children": [
              {
                "title": "Namespaces",
                "children": [
                  {
                    "title": "NestedNamespace",
                    "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "title": "index",
    "url": "index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for multiple entry points: (Output File Strategy "modules") (Option Group "2") 1`] = `
"[
  {
    "title": "API",
    "url": "index.md"
  },
  {
    "title": "basic",
    "url": "basic.md"
  },
  {
    "title": "has-categories",
    "url": "has-categories.md",
    "children": []
  },
  {
    "title": "has-namespaces",
    "url": "has-namespaces/index.md",
    "children": [
      {
        "title": "NamespaceB",
        "url": "has-namespaces/namespaces/NamespaceB.md"
      },
      {
        "title": "_Namespace_A_",
        "url": "has-namespaces/namespaces/Namespace_A_/index.md",
        "children": [
          {
            "title": "NestedNamespace",
            "url": "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md"
          }
        ]
      }
    ]
  },
  {
    "title": "index",
    "url": "module.index.md"
  }
]"
`;

exports[`Navigation should gets Navigation Json for single entry point: (Output File Strategy "members") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "exports.md"
  },
  {
    "title": "Enumerations",
    "children": [
      {
        "title": "BasicEnum",
        "url": "enumerations/BasicEnum.md"
      },
      {
        "title": "EnumWithValues",
        "url": "enumerations/EnumWithValues.md"
      }
    ]
  },
  {
    "title": "Classes",
    "children": [
      {
        "title": "AbstractClass",
        "url": "classes/AbstractClass.md"
      },
      {
        "title": "ClassWithAccessors",
        "url": "classes/ClassWithAccessors.md"
      },
      {
        "title": "ClassWithConstructorOverloads",
        "url": "classes/ClassWithConstructorOverloads.md"
      },
      {
        "title": "ClassWithModifiers",
        "url": "classes/ClassWithModifiers.md"
      },
      {
        "title": "ClassWithTypeParameters",
        "url": "classes/ClassWithTypeParameters.md"
      },
      {
        "title": "DerivedClassA",
        "url": "classes/DerivedClassA.md"
      },
      {
        "title": "DerivedClassB",
        "url": "classes/DerivedClassB.md"
      }
    ]
  },
  {
    "title": "Interfaces",
    "children": [
      {
        "title": "BasicInterface",
        "url": "interfaces/BasicInterface.md"
      },
      {
        "title": "ExtendedInterface",
        "url": "interfaces/ExtendedInterface.md"
      },
      {
        "title": "IndexableInterface",
        "url": "interfaces/IndexableInterface.md"
      },
      {
        "title": "InterfaceWithTypeParameters",
        "url": "interfaces/InterfaceWithTypeParameters.md"
      }
    ]
  },
  {
    "title": "Type Aliases",
    "children": [
      {
        "title": "ArrayType",
        "url": "type-aliases/ArrayType.md"
      },
      {
        "title": "ConditionalType",
        "url": "type-aliases/ConditionalType.md"
      },
      {
        "title": "ExternalReferenceType",
        "url": "type-aliases/ExternalReferenceType.md"
      },
      {
        "title": "FunctionType",
        "url": "type-aliases/FunctionType.md"
      },
      {
        "title": "IntersectionType",
        "url": "type-aliases/IntersectionType.md"
      },
      {
        "title": "LiteralType",
        "url": "type-aliases/LiteralType.md"
      },
      {
        "title": "PartialMappedType",
        "url": "type-aliases/PartialMappedType.md"
      },
      {
        "title": "PrimitiveType",
        "url": "type-aliases/PrimitiveType.md"
      },
      {
        "title": "ReadonlyMappedType",
        "url": "type-aliases/ReadonlyMappedType.md"
      },
      {
        "title": "StringLiteralType",
        "url": "type-aliases/StringLiteralType.md"
      },
      {
        "title": "TupleType",
        "url": "type-aliases/TupleType.md"
      },
      {
        "title": "TypeWithExternalSymbolLinkMapping",
        "url": "type-aliases/TypeWithExternalSymbolLinkMapping.md"
      },
      {
        "title": "TypeWithTypeParams",
        "url": "type-aliases/TypeWithTypeParams.md"
      },
      {
        "title": "UnionType",
        "url": "type-aliases/UnionType.md"
      }
    ]
  },
  {
    "title": "Variables",
    "children": [
      {
        "title": "objectLiteralVariable",
        "url": "variables/objectLiteralVariable.md"
      },
      {
        "title": "stringVariable",
        "url": "variables/stringVariable.md"
      },
      {
        "title": "typeOperatorVariable",
        "url": "variables/typeOperatorVariable.md"
      }
    ]
  },
  {
    "title": "Functions",
    "children": [
      {
        "title": "basicFunction",
        "url": "functions/basicFunction.md"
      },
      {
        "title": "functionReturningAFunction",
        "url": "functions/functionReturningAFunction.md"
      },
      {
        "title": "functionReturningAPromise",
        "url": "functions/functionReturningAPromise.md"
      },
      {
        "title": "functionReturningAnObject",
        "url": "functions/functionReturningAnObject.md"
      },
      {
        "title": "functionWithDefaultParameters",
        "url": "functions/functionWithDefaultParameters.md"
      },
      {
        "title": "functionWithMultipleSignatures",
        "url": "functions/functionWithMultipleSignatures.md"
      },
      {
        "title": "functionWithNamedParams",
        "url": "functions/functionWithNamedParams.md"
      },
      {
        "title": "functionWithNestedParameters",
        "url": "functions/functionWithNestedParameters.md"
      },
      {
        "title": "functionWithOptionalParameters",
        "url": "functions/functionWithOptionalParameters.md"
      },
      {
        "title": "functionWithTypeParameters",
        "url": "functions/functionWithTypeParameters.md"
      }
    ]
  }
]"
`;

exports[`Navigation should gets Navigation Json for single entry point: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[
  {
    "title": "README",
    "url": "README.md"
  },
  {
    "title": "API",
    "url": "exports.md"
  }
]"
`;
